lib/Fmt_ast.ml

```diff
@@ -1528,11 +1528,18 @@ and fmt_infix_op_args c ~parens xexp op_args =
       | `Closing_on_separate_line -> ("(", Some (1000, 0), ")")
     else ("", None, "")
   in
-  wrap_if_k
-    (parens || Ast.parenze_nested_exp xexp)
-    (fits_breaks "(" opn)
-    (fits_breaks ")" ?hint cls)
-    (list_fl groups fmt_op_arg_group)
+  match
+    Params.parens_or_begin_end c.conf c.source ~loc:xexp.ast.pexp_loc
+  with
+  | `Parens ->
+      wrap_if_k
+        (parens || Ast.parenze_nested_exp xexp)
+        (fits_breaks "(" opn)
+        (fits_breaks ")" ?hint cls)
+        (list_fl groups fmt_op_arg_group)
+  | `Begin_end ->
+      wrap "begin@;<1 2>@[<0>" "@]@;<1000 0>end"
+        (list_fl groups fmt_op_arg_group)
```

Comment:
Can't this be calling `wrap_exp` below ?

Reply:
I've done that now, a few changes in parens, but the code is much easier to maintain, and some cases of breaking the margin have been solved in the testbase.

Reply:
Right, so I think it's better to just keep the first commit, wrap_exp does not fit this usecase

Reply:
What's the reason ?

Reply:
The code was too different from `wrap_exp`, I tried to merge both into one, but there was arguments/options only used for this "infix_op_arg" usecase and others only used for the general case, so in my last commit I chose to rename `Params.wrap_exp` as `Params.Exp.wrap` and define a new function `Params.Exp.Infix_op_arg.wrap` to use here (in future PRs I would like to put more expression-related functions into the Exp module)